kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/internal/parquet/ParquetFileReaderSuite.scala

```diff
@@ -0,0 +1,276 @@
+
+import scala.collection.mutable.ArrayBuffer
+
+import io.delta.golden.GoldenTableUtils.{goldenTableFile, goldenTablePath}
+import io.delta.kernel.defaults.internal.DefaultKernelUtils.DateTimeConstants
+import io.delta.kernel.defaults.utils.{ExpressionTestUtils, TestRow, VectorTestUtils}
+import io.delta.kernel.types._
+import org.scalatest.funsuite.AnyFunSuite
+
+class ParquetFileReaderSuite extends AnyFunSuite
+  with ParquetSuiteBase with VectorTestUtils with ExpressionTestUtils {
+
+  def getSingleParquetFile(directory: File): String = {
+    val parquetFiles = directory.listFiles().filter(_.getName.endsWith(".parquet"))
+    assert(parquetFiles.size == 1)
+    parquetFiles.head.getAbsolutePath
+  }
+
+  //////////////////////////////////////////////////////////////////////////////////
+  // Decimal type tests
+  //////////////////////////////////////////////////////////////////////////////////
+
+  private val DECIMAL_TYPES_DICT_FILE_V1 = getSingleParquetFile(
+    goldenTableFile("parquet-decimal-dictionaries-v1"))
+
+  private val DECIMAL_TYPES_DICT_FILE_V2 = getSingleParquetFile(
+    goldenTableFile("parquet-decimal-dictionaries-v2"))
+
+  test("decimals encoded using dictionary encoding ") {
+    val expResult = (0 until 1000000).map { i =>
+      TestRow(i, BigDecimal.valueOf(i%5), BigDecimal.valueOf(i%6), BigDecimal.valueOf(i%2))
+    }
+
+    val readSchema = new StructType()
+      .add("id", IntegerType.INTEGER)
+      .add("col1", new DecimalType(9, 0)) // INT32: 1 <= precision <= 9
+      .add("col2", new DecimalType(12, 0)) // INT64: 10 <= precision <= 18
+      .add("col3", new DecimalType(25, 0)) // FIXED_LEN_BYTE_ARRAY
+
+    for (file <- Seq(DECIMAL_TYPES_DICT_FILE_V1, DECIMAL_TYPES_DICT_FILE_V2)) {
+      val actResult = readParquetFilesUsingSpark(file, readSchema)
+
+      checkAnswer(actResult, expResult)
+    }
+  }
+
+  private val LARGE_SCALE_DECIMAL_TYPES_FILE = getSingleParquetFile(
+    goldenTableFile("parquet-decimal-type"))
+
+  test("large scale decimal type file") {
+
+    def expand(n: BigDecimal): BigDecimal = {
+      n.scaleByPowerOfTen(5).add(n)
+    }
+
+    val expResult = (0 until 99998).map { i =>
+      if (i % 85 == 0) {
+        val n = BigDecimal.valueOf(i)
+        TestRow(i, n.movePointLeft(1).setScale(1), n.setScale(5), n.setScale(5))
+      } else {
+        val negation = if (i % 33 == 0) {
+          -1
+        } else {
+          1
+        }
+        val n = BigDecimal.valueOf(i*negation)
+        TestRow(
+          i,
+          n.movePointLeft(1),
+          expand(n).movePointLeft(5),
+          expand(expand(expand(n))).movePointLeft(5)
+        )
+      }
+    }
+
+    val readSchema = new StructType()
+      .add("id", IntegerType.INTEGER)
+      .add("col1", new DecimalType(5, 1)) // INT32: 1 <= precision <= 9
+      .add("col2", new DecimalType(10, 5)) // INT64: 10 <= precision <= 18
+      .add("col3", new DecimalType(20, 5)) // FIXED_LEN_BYTE_ARRAY
```

Comment:
you can use the `tableSchema` API to fetch the schema. Keep/update the comment mentioning three different types of storage for decimal columns